patches/minecraft/net/minecraft/client/color/item/ItemColors.java.patch

```diff
@@ -10,6 +10,15 @@
  
     public static ItemColors m_92683_(BlockColors p_92684_) {
        ItemColors itemcolors = new ItemColors();
+@@ -68,7 +_,7 @@
+          return p_92700_ > 0 ? -1 : PotionUtils.m_43575_(p_92699_);
+       }, Items.f_42589_, Items.f_42736_, Items.f_42739_);
+ 
+-      for(SpawnEggItem spawneggitem : SpawnEggItem.m_43233_()) {
++      for(SpawnEggItem spawneggitem : net.minecraftforge.common.ForgeSpawnEggItem.eggs()) {
+          itemcolors.m_92689_((p_92681_, p_92682_) -> {
```

Comment:
Instead of patching this, you should just use the event we have already, and leave the vanilla path to itself.

Reply:
Should I move this to a Forge internal handler with `HIGHEST` priority to make sure mods can still override the color handler with their own or leave it to the mod entirely?

Reply:
I am not a fan of helper methods for no good reason. So I'm more leaning twards let the modders do it.
However I know people are lazy about it. So yes, move it to an event handler.
Honestly this can all be wrapped in one class, with no access or patch changes.
And one line edited in IForgeEntity.getPickedResult.

Reply:
Using our own map, not moving stuff to the `BY_ID` map and changing `IForgeEntity.getPickedResult()` would still result in having to use the additional list and moving stuff over at a certain point.

Removing the deprecation patch to `SpawnEggItem` would be counterproductive IMO because no one would care about the replacement because there is no IDE warning that annoys them.
The other patch line in `SpawnEggItem` is intended to avoid adding a null key to the `BY_ID` map which would lead to a random spawn egg from a random mod being duplicated and getting a color handler registered for it twice.

Reply:
>Using our own map, not moving stuff to the BY_ID map and changing IForgeEntity.getPickedResult() would still result in having to use the additional list and moving stuff over at a certain point.

Why?

Reply:
Because we can't build a type->item map when the items are constructed, that must always happen after entity type registration. Building a reverse map instead is pointless because we would need to iterate all items until we find the correct one, might as well only use a list in that case.

Reply:
Moved the color handler registration to an event handler